### PR TITLE
[DOC-8977] Add step to configure systemd to start CockroachDB automatically after a reboot

### PR DESCRIPTION
**Pull request by mdlinville**
[DOC-8977] Add step to configure systemd to start CockroachDB automatically after a reboot

These includes touch several files each in each of v24.1, v23.2, and v23.1. For HTML previews, refer to https://github.com/cockroachdb/docs/pull/18439#issuecomment-2033172706.

### CODE REVIEW

**Review by jhatcher9999 — COMMENTED** [review]
LGTM

File: src/current/v23.1/recommended-production-settings.md [hunk 1]
```diff
@@ -577,65 +582,71 @@ For example, for a node with 3 stores, we would set the hard limit to at least 3
     session    required   pam_limits.so
     ~~~
 
-1.  Edit `/etc/security/limits.conf` and append the following lines to the file:
+1.  Set a limit for the number of open file descriptors. The specific limit you set depends on your workload and the hardware and configuration of your nodes.
```
Comment (mdlinville) [inline src/current/v23.1/recommended-production-settings.md:585]: The changes to this file look much larger than they actually are. This content was refactored for 23.2+ at some point and 23.1 didn't get that refactor. This brings the 23.1 section back in line with 23.2+.

File: src/current/v23.1/recommended-production-settings.md [hunk 2]
```diff
@@ -565,6 +565,11 @@ For example, for a node with 3 stores, we would set the hard limit to at least 3
 - [Per-Process Limit](#per-process-limit)
 - [System-Wide Limit](#system-wide-limit)
 
+<section id="linuxinstall" markdown="1">
+
+- [Per-Process Limit](#per-process-limit)
+- [System-Wide Limit](#system-wide-limit)
+
```
Comment (kathancox) [inline src/current/v23.1/recommended-production-settings.md:572]: There's a weird double-up here and render problem for one of the sets.

File: src/current/v23.1/recommended-production-settings.md [hunk 3]
```diff
@@ -577,65 +582,71 @@ For example, for a node with 3 stores, we would set the hard limit to at least 3
...
     ~~~
 
-1.  Edit `/etc/security/limits.conf` and append the following lines to the file:
+1.  Set a limit for the number of open file descriptors. The specific limit you set depends on your workload and the hardware and configuration of your nodes.
 
-    ~~~
-    *              soft     nofile          35000
-    *              hard     nofile          35000
-    ~~~
+    - **If you use `systemd`**, manually-set limits set using the `ulimit` command or a configuration file like `/etc/limits.conf` are ignored for services started by `systemd`. To limit the number of open file descriptors, add a line like the following to the service definition for the `cockroach` process. To allow an unlimited number of files, you can optionally set `LimitNOFILE` to `INFINITY`. Cockroach Labs recommends that you carefully test this configuration with a realistic workload before deploying it in production.
```
Comment (kathancox) [inline src/current/v23.1/recommended-production-settings.md:587]: From here, in v23.1 + v24.1 the numbered list and bullets are not rendering. (From numbered list item 2.)

Comment (kathancox) [reply src/current/v23.1/recommended-production-settings.md:587]: (It is rendering fine in v23.2)

[the remark this answers is not in the document]
Comment (kathancox) [reply src/current/v23.1/recommended-production-settings.md:587]: This is a suggestion — I found the double "set" in that sentence as a modifier and verb made me re-read the sentence.

**Review by kathancox — COMMENTED** [review]
For the production settings file, there is a render problem for two of the versions (see inline comments). Lmk when you want me to look again, thanks!

**Review by kathancox — APPROVED** [review]
LGTM!

**Pull request merged**